src/commands/editor.rs

```diff
@@ -0,0 +1,257 @@
+use std::collections::HashSet;
+use std::fs::File;
+use std::path::Path;
+use std::io::{stdout, BufWriter, Write};
+use std::path::PathBuf;
+
+use anyhow::{bail, Result, ensure};
+use serde::{Deserialize, Serialize};
+
+use hdr10plus::metadata_json::{Hdr10PlusJsonMetadata, MetadataJsonRoot};
+
+use crate::commands::EditorArgs;
+
+pub const TOOL_NAME: &str = env!("CARGO_PKG_NAME");
+pub const TOOL_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+use super::input_from_either;
+
+pub struct Editor {
+    edits_json: PathBuf,
+    json_out: PathBuf,
+
+    metadata_root: MetadataJsonRoot,

```

Comment:
The editing should be working on a list of `Hdr10PlusMetadata`, not the JSON root.
```rust
let metadata_list: Vec<Hdr10PlusMetadata> = metadata_root
    .scene_info
    .iter()
    .map(Hdr10PlusMetadata::try_from)
    .filter_map(Result::ok)
    .collect();
```

Because currently you're having to deal with scenes and re-fixing the numbers when the logic already exists.
```rust
pub fn generate_json(
    metadata: &[&Hdr10PlusMetadata],
    tool_name: &str,
    tool_version: &str,
)
```

Which already computes the scene info.